main.go

```diff
@@ -152,113 +145,11 @@ var webSocketUpgrade = websocket.Upgrader{
 	},
 }
 
-func getToken(c *gin.Context) (*jwt.Token, error) {
-	tokenString := c.GetHeader("Authorization")
-
-	if tokenString == "" {
-		origin := c.GetHeader("Origin")
-		c.JSON(http.StatusForbidden, gin.H{"err": "token is required"})
-		log.Error(fmt.Sprintf("Attempt to access resources without token from origin: %s", origin))
-		return nil, errors.New("Error")
-	}
-
-	tokenString = tokenString[len("Bearer "):]
-
-	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
-		return verifyKey, nil
-	})
-
-	return token, err
-}
-
-func userIsLoggedIn() gin.HandlerFunc {
-	return func(c *gin.Context) {
-
-		token, err := getToken(c)
-
-		if err != nil {
-			c.Abort()
-			return
-		}
-
-		if token.Valid {
-			// save token in context for use in other middleware
-			c.Set("token", token)
-		} else {
-			origin := c.GetHeader("Origin")
-			if ve, ok := err.(*jwt.ValidationError); ok {
-				if ve.Errors&jwt.ValidationErrorMalformed != 0 {
-					c.JSON(http.StatusForbidden, gin.H{"err": "token is malformed"})
-					log.Error(fmt.Sprintf("Attempt to access resources with malformed token from origin: %s", origin))
-				} else if ve.Errors&(jwt.ValidationErrorExpired) != 0 {
-					// Token is expired
-					c.JSON(http.StatusForbidden, gin.H{"err": "token is expired"})
-				} else {
-					c.JSON(http.StatusForbidden, gin.H{"err": "error reading token"})
-					log.Error(fmt.Sprintf("Error accessing resources with token from origin %s", origin))
-				}
-			} else {
-				c.JSON(http.StatusForbidden, gin.H{"err": "error reading token"})
-				log.Error(fmt.Sprintf("Error accessing resources with token from origin %s", origin))
-			}
-			c.Abort()
-		}
-	}
-}
-
-func userIsInRole(d database.IDatabase, roles []constants.Role) gin.HandlerFunc {
-	return func(c *gin.Context) {
-		var token interface{}
-		var ok bool
-
-		if token, ok = c.Get("token"); !ok {
-			c.JSON(http.StatusForbidden, gin.H{"err": "Error accessing token"})
-			c.Abort()
-			return
-		}
-
-		var userRole constants.Role
-		if claims, ok := token.(*jwt.Token).Claims.(jwt.MapClaims); ok {
-			userRole = constants.Role(claims["role"].(float64))
-		} else {
-			c.Abort()
-			return
-		}
-
-		for _, role := range roles {
-			if userRole == role {
-				return
-			}
-		}
-
-		c.JSON(http.StatusForbidden, gin.H{"err": "User doesn't have access"})
-		c.Abort()
-	}
-}
-
 func init() {
-	privKeyPath := os.Getenv("PRIVATE_KEY_PATH")
-	pubKeyPath := os.Getenv("PUBLIC_KEY_PATH")
+	auth := auth.Auth{}
```

Comment:
Might want to do `au := auth.Auth{}` and `a = &au` just to get rid of any name overloading